src/CalcViewModel/UnitConverterViewModel.cpp

```diff
@@ -782,13 +782,16 @@ void UnitConverterViewModel::RefreshSupplementaryResults()
 // values are not both zero.
 void UnitConverterViewModel::AnnounceConversionResult()
 {
-    if ((m_valueFromUnlocalized != m_lastAnnouncedFrom || m_valueToUnlocalized != m_lastAnnouncedTo) && Unit1 != nullptr && Unit2 != nullptr)
+    if ((m_valueFromUnlocalized != m_lastAnnouncedFrom
+        || m_valueToUnlocalized != m_lastAnnouncedTo)
+        && m_Unit1 != nullptr
+        && m_Unit2 != nullptr)
```

Comment:
the method header mentions a validation against zero as well, does zero need to be added as a conditional to the if?